src/item_location.cpp

```diff
@@ -478,6 +482,94 @@ class item_location::impl::item_on_vehicle : public item_location::impl
         }
 };
 
+class item_location::impl::item_in_container : public item_location::impl
+{
+    private:
+        item_location container;
+
+        // figures out the index for the item, which is where it is in the total list of contents
+        // note: could be a better way of handling this?
+        int calc_index() const {
+            int idx = 0;
+            for( const item *it : container->contents.all_items_ptr() ) {
+                if( target() == it ) {
+                    return idx;
+                }
+                idx++;
+            }
+            if( idx == 0 ) {
+                return -1;
+            }
+            return idx;
+        }
+    public:
+        item_location parent_item() const override {
+            return container;
+        }
+
+        item_in_container( const item_location &container, item *which ) :
+            impl( which ), container( container ) {}
+
+        void serialize( JsonOut &js ) const override {
+            js.start_object();
+            js.member( "idx", calc_index() );
+            js.member( "type", "in_container" );
```

Comment:
You need additional code in `item_location::deserialize` to handle the new type.